### ISSUE TITLE
Logging for ImportVm

### ISSUE THREAD

**Issue by rajujith** [apache/cloudstack#8779]
##### ISSUE TYPE

 * Improvement Request
 

##### COMPONENT NAME

~~~
Logging
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### CONFIGURATION

External KVM with VMs for import
Cloudstack with KVM cluster

##### SUMMARY

The logging for ImportVm, specifically importing instance from external KVM appears to be unusable. The Command sequence send to the KVM agent and the answer are hidden in TRACE level logging. Although many of the logging statements are DEBUG they don't appear unless TRACE is enabled. 
Only 'CopyRemoteVolumeAnswer' is seen once in the logs if we follow the async job id. The implementation is same for KVM agent as well. Improving the logging for this operation to match all other operations in Cloudstack will be convenient for operation and maintenance. 

Seq send to the agent such as the below are missing in the DEBUG level which is not expected. 
com.cloud.agent.api.GetRemoteVmsCommand
com.cloud.agent.api.GetRemoteVmsAnswer
com.cloud.agent.api.CopyRemoteVolumeCommand

**Comment by rajujith** [comment]
In addition  to the above some logs are TRACE for no reason
```
2024-03-13 07:19:27,212 TRACE [o.a.c.a.c.a.v.ImportUnmanagedInstanceCmd] (API-Job-Executor-28:ctx-4aba9ac4 job-34 ctx-82af2de0) (logid:c6e80df2) nic, '0', goes on net, 'fe6ec93e-6c9e-4e23-b879-6cc70390c0b5'
2024-03-13 07:19:27,213 TRACE [o.a.c.a.c.a.v.ImportUnmanagedInstanceCmd] (API-Job-Executor-28:ctx-4aba9ac4 job-34 ctx-82af2de0) (logid:c6e80df2) nic, '0', gets ip, 'auto'
```

**Comment by DaanHoogland** [comment]
@rajujith , TRACE are set on purpose by a developer, so that must be for a reason. I think nothing below INFO should be visible by default. 

DEBUG not be visible beyond TRACE is a **bug**, however!

> Although many of the logging statements are DEBUG they don't appear unless TRACE is enabled.

Do you have pointers on which DEBUG has this problem?

so in summary, you want the sending of
- com.cloud.agent.api.GetRemoteVmsCommand
- com.cloud.agent.api.GetRemoteVmsAnswer
- com.cloud.agent.api.CopyRemoteVolumeCommand
to be logged as DEBUG?

is there more?

**Comment by rajujith** [comment]
@DaanHoogland  I want reasonable logging for the ImportVM operation in the DEBUG level for better serviceability. This includes making sure the DEBUG logs for the mentioned commands are logged without having to set the logging verbosity to TRACE level.

**Comment by DaanHoogland** [comment]
@vishesh92 , you assigned yourself, do you have any implementation plan? (i moved this out of 4.19.1!)

**Comment by vishesh92** [comment]
> @vishesh92 , you assigned yourself, do you have any implementation plan? (i moved this out of 4.19.1!)

I don't have any plan as of now.

**Comment by sureshanaparti** [comment]
Fixed in #9284
